src/eei.cpp

```diff
@@ -920,7 +920,7 @@ inline int64_t maxCallGas(int64_t gas) {
     ensureCondition(memorySize() >= (offset + length), InvalidMemoryAccess, "Out of bounds (source) memory copy.");
   }
 
-  void EthereumInterface::loadMemory(uint32_t srcOffset, uint8_t *dst, size_t length)
+  void EthereumInterface::loadMemoryReverse(uint32_t srcOffset, uint8_t *dst, size_t length)
```

Comment:
Perhaps it would make sense to explicitly annotate the function name, e.g. BE2LE
Or do the c++ thing and have an `IOManager` class wrapping around the ShellInterface with templated functions for performing the access operations with or without endianness swapping.

In general, the data flow in and out of EVMC and the WASM module could be better represented with another abstraction layer.

Reply:
Well this is the first step to make it apparent. Then I think we need to review it and ensure that byteswapping only happens on numeric types (practically only balance). Storage should not be swapped.